src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/EnvelopeProcessor.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services;
+
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageReceiver;
+import com.microsoft.azure.servicebus.primitives.ServiceBusException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.client.BulkScanProcessorClient;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.exceptions.ReadEnvelopeException;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.bulkscanprocessorclient.model.Envelope;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.ReceiverProvider;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.exceptions.ConnectionException;
+
+public class EnvelopeProcessor {
+    private static final Logger logger = LoggerFactory.getLogger(EnvelopeProcessor.class);
+
+    private final ReceiverProvider receiverProvider;
+    private final BulkScanProcessorClient bulkScanProcessorClient;
+
+    public EnvelopeProcessor(
+        ReceiverProvider receiverProvider,
+        BulkScanProcessorClient bulkScanProcessorClient
+    ) {
+        this.receiverProvider = receiverProvider;
+        this.bulkScanProcessorClient = bulkScanProcessorClient;
+    }
+
+    public void run() {
```

Comment:
Should we run this on schedule?

Reply:
It would be good to add some non-error logging, so that we can see that a message was read from the queue, for example.

Reply:
I've planned running this on a schedule, making it configurable etc for next PR.